### PR TITLE
TST: add test to check dtype after replacing values in categorical Series inplace

### LINKED ISSUE #53358: BUG: `Series.replace` doesn't update `CategoricalDtype` when `inplace=True` in pandas-2.0
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
In [1]: import pandas as pd

In [2]: s = pd.Series(["one", "two", "three"], dtype="category")

In [3]: s
Out[3]: 
0      one
1      two
2    three
dtype: category
Categories (3, object): ['one', 'three', 'two']

In [4]: new_s = s.replace(to_replace="one", value="two")

In [5]: new_s
Out[5]: 
0      two
1      two
2    three
dtype: category
Categories (2, object): ['three', 'two']

In [6]: new_s.dtype
Out[6]: CategoricalDtype(categories=['three', 'two'], ordered=False)     # Correct

In [7]: s.replace(to_replace="one", value="two", inplace=True)

In [8]: s
Out[8]: 
0      two
1      two
2    three
dtype: category
Categories (2, object): ['three', 'two']          # Correct __repr__ output

In [9]: s.dtype
Out[9]: CategoricalDtype(categories=['one', 'three', 'two'], ordered=False). # BUG: Incorrect dtype, dtype used to be updated pre pandas-2.0
```

### Issue Description

When we replace values in a categorical Series, with `inplace=True`, the dtype is not being updated.

### Expected Behavior

```python
In [9]: s.dtype
Out[9]: CategoricalDtype(categories=['three', 'two'], ordered=False)
```

### Installed Versions

<details>

INSTALLED VERSIONS
------------------
commit           : 37ea63d540fd27274cad6585082c91b1283f963d
python           : 3.10.11.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.15.0-76-generic
Version          : #86-Ubuntu SMP Fri Jan 17 17:24:28 UTC 2020
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : en_US.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 2.0.1
numpy            : 1.23.5
pytz             : 2023.3
dateutil         : 2.8.2
setuptools       : 67.7.2
pip              : 23.1.2
Cython           : 0.29.34
pytest           : 7.3.1
hypothesis       : 6.75.3
sphinx           : 5.3.0
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : 3.1.2
IPython          : 8.13.2
pandas_datareader: None
bs4              : 4.12.2
bottleneck       : None
brotli           : 
fastparquet      : None
fsspec           : 2023.5.0
gcsfs            : None
matplotlib       : None
numba            : 0.56.4
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : 11.0.0
pyreadstat       : None
pyxlsb           : None
s3fs             : 2023.5.0
scipy            : 1.10.1
snappy           : 
sqlalchemy       : 2.0.15
tables           : None
tabulate         : 0.9.0
xarray           : None
xlrd             : None
zstandard        : None
tzdata           : 2023.3
qtpy             : None
pyqt5            : None

</details>

### PR DESCRIPTION
**Pull request by natmokval**
- [x] closes #53358

- added a test in order to check if the `dtype` is being updated, when we replace values in a categorical Series with `inplace=True`.

### CODE REVIEW

**Comment by natmokval** [comment]
@mroeschke, I added a test to check if the `dtype` is being updated, when we replace values in a categorical Series `inplace`. Could you please take a look at my pr?

File: pandas/tests/series/methods/test_replace.py [hunk 1]
```diff
@@ -387,6 +387,16 @@ def test_replace_categorical(self, categorical, numeric):
...
 
+    @pytest.mark.parametrize(
+        "data, data_exp", [(["a", "b", "c"], ["b", "b", "c"]), (["a"], ["b"])]
+    )
+    def test_replace_categorical_inplace(self, data, data_exp):
+        # GH 53358
+        result = pd.Series(data, dtype="category")
+        result.replace(to_replace="a", value="b", inplace=True)
+        expected = pd.Series(data_exp, dtype="category")
+        tm.assert_index_equal(expected.dtype.categories, result.dtype.categories)
```
Comment (phofl) [inline pandas/tests/series/methods/test_replace.py:398]: Is this not covered by tm.assert_series_equal?

Comment (natmokval) [reply pandas/tests/series/methods/test_replace.py]: Thank you, I replaced `tm.assert_index_equal` with `tm.assert_series_equal`.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @natmokval

**Pull request merged**